File: client/main/Assets/Program/Scripts/Msg/beans/play/wordboss/BossRankInfo.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DreamFaction.GameNetWork.Data;

namespace GNET
{
    public class BossRankInfo : Marshal
	{
        public long roleid; // Íæ¼Òguid
        public string rolename; // Íæ¼ÒÃû³Æ
        public long num; // ÉËº¦

        public BossRankInfo()
        {
            roleid = 0;
            rolename = "";
            num = 0;
        }

        public BossRankInfo(long _roleid_, string _rolename_, long _num_)
        {
            this.roleid = _roleid_;
            this.rolename = _rolename_;
            this.num = _num_;
        }

        public override OctetsStream marshal(OctetsStream _os_)
		{
            _os_.marshal(roleid);
            _os_.marshal(rolename);
            _os_.marshal(num);
    		return _os_;
		}

        public override OctetsStream unmarshal(OctetsStream _os_)
		{
            roleid = _os_.unmarshal_long();
            rolename = _os_.unmarshal_String();
            num = _os_.unmarshal_long();
            return _os_;
		}
	}
}

File: Server/src/gs/beans/chuhan/gsp/play/wordboss/BossRankInfo.java

package chuhan.gsp.play.wordboss;

import com.goldhuman.Common.Marshal.Marshal;
import com.goldhuman.Common.Marshal.OctetsStream;
import com.goldhuman.Common.Marshal.MarshalException;

public class BossRankInfo implements Marshal {
	public long roleid; // 玩家guid
	public java.lang.String rolename; // 玩家名称
	public long num; // 伤害

	public BossRankInfo() {
		rolename = "";
	}

	public BossRankInfo(long _roleid_, java.lang.String _rolename_, long _num_) {
		this.roleid = _roleid_;
		this.rolename = _rolename_;
		this.num = _num_;
	}

	public final boolean _validator_() {
		return true;
	}

	public OctetsStream marshal(OctetsStream _os_) {
		_os_.marshal(roleid);
		_os_.marshal(rolename, "UTF-16LE");
		_os_.marshal(num);
		return _os_;
	}

	public OctetsStream unmarshal(OctetsStream _os_) throws MarshalException {
		roleid = _os_.unmarshal_long();
		rolename = _os_.unmarshal_String("UTF-16LE");
		num = _os_.unmarshal_long();
		return _os_;
	}

	public boolean equals(Object _o1_) {
		if (_o1_ == this) return true;
		if (_o1_ instanceof BossRankInfo) {
			BossRankInfo _o_ = (BossRankInfo)_o1_;
			if (roleid != _o_.roleid) return false;
			if (!rolename.equals(_o_.rolename)) return false;
			if (num != _o_.num) return false;
			return true;
		}
		return false;
	}

	public int hashCode() {
		int _h_ = 0;
		_h_ += (int)roleid;
		_h_ += rolename.hashCode();
		_h_ += (int)num;
		return _h_;
	}

	public String toString() {
		StringBuilder _sb_ = new StringBuilder();
		_sb_.append("(");
		_sb_.append(roleid).append(",");
		_sb_.append("T").append(rolename.length()).append(",");
		_sb_.append(num).append(",");
		_sb_.append(")");
		return _sb_.toString();
	}

}

